components/ai_chat/core/common/proto_conversion_unittest.cc

```diff
@@ -122,12 +142,132 @@ TEST(ProtoConversionTest, SerializeDeserializeWebSourcesEvent_EmptySources) {
 
   // Verify empty proto
   EXPECT_EQ(proto_event.sources_size(), 0);
+  EXPECT_EQ(proto_event.rich_results_size(), 0);
 
   // Deserialize back to mojom
   auto deserialized_event = DeserializeWebSourcesEvent(proto_event);
 
   // Verify empty mojom
   EXPECT_EQ(deserialized_event->sources.size(), 0u);
+  EXPECT_EQ(deserialized_event->rich_results.size(), 0u);
+}
+
+TEST(ProtoConversionTest, SerializeDeserializeWebSourcesEvent_RichResultsOnly) {
+  // Create mojom WebSourcesEvent with only rich_results, no sources
+  auto mojom_event = mojom::WebSourcesEvent::New();
+
+  // Add rich_results without sources
+  mojom_event->rich_results.push_back(
+      R"({"type":"answer_box","answer":"42","question":"meaning of life"})");
+  mojom_event->rich_results.push_back(
+      R"({"type":"calculator","result":"123","expression":"100+23"})");
+  mojom_event->rich_results.push_back(
+      R"({"type":"weather","temperature":"72F","location":"San Francisco"})");
+
+  // Serialize to proto
+  store::WebSourcesEventProto proto_event;
+  SerializeWebSourcesEvent(mojom_event, &proto_event);
+
+  // Verify proto data
+  EXPECT_EQ(proto_event.sources_size(), 0);
+  ASSERT_EQ(proto_event.rich_results_size(), 3);
+  EXPECT_EQ(
+      proto_event.rich_results(0),
+      R"({"type":"answer_box","answer":"42","question":"meaning of life"})");
+  EXPECT_EQ(proto_event.rich_results(1),
+            R"({"type":"calculator","result":"123","expression":"100+23"})");
+  EXPECT_EQ(
+      proto_event.rich_results(2),
+      R"({"type":"weather","temperature":"72F","location":"San Francisco"})");
+
+  // Deserialize back to mojom
+  auto deserialized_event = DeserializeWebSourcesEvent(proto_event);
+
+  // Verify deserialized data
+  EXPECT_EQ(deserialized_event->sources.size(), 0u);
+  ASSERT_EQ(deserialized_event->rich_results.size(), 3u);
+  EXPECT_EQ(
+      deserialized_event->rich_results[0],
+      R"({"type":"answer_box","answer":"42","question":"meaning of life"})");
+  EXPECT_EQ(deserialized_event->rich_results[1],
+            R"({"type":"calculator","result":"123","expression":"100+23"})");
+  EXPECT_EQ(
+      deserialized_event->rich_results[2],
+      R"({"type":"weather","temperature":"72F","location":"San Francisco"})");
+}
+
+TEST(ProtoConversionTest,
+     SerializeDeserializeWebSourcesEvent_ComplexRichResults) {
+  // Test with complex nested JSON in rich_results
```

Comment:
Is there any point having this different test - what makes the JSON more complex? Could we just add the complex JSON string to the previous test?

Reply:
removed the other one